### DISCUSSION TITLE
Delete contact and chat messages

### DISCUSSION THREAD

**Discussion by mrjackson6969** [oxen-io/session-desktop#2933] [Q&A] [answered]
I run app on Ubuntu 22.04.3
On right click on person i have:
 -Delete Messages
 -Delete Conversation
 -Delete Contact
Which one of those suppose to delete messages on both sides? All on my side and those I sent on the other side? From my experience none work. I don't know, I tried to used them all, and all it does it to delete messages in my app but other people can still see the old chat. And when i recover account on new device I have access to old, deleted messages.

**Comment by KeeJef** [comment, marked as answer]
Right now the only way to delete messages for yourself and your chat partner/s is to right click on a message and select "Delete for everyone" or select messages and "Delete for everyone". There is no option to right click on a conversation and delete all messages for everyone

**Reply by mrjackson6969** [reply]
What is the difference between "Delete contact" and "Delete conversation"? It both look like it deletes messages from the device but not from servers.
